pkg/jobrunaggregator/jobrunhistoricaldataanalyzer/analyzer.go

```diff
@@ -91,6 +91,30 @@ func (o *JobRunHistoricalDataAnalyzerOptions) Run(ctx context.Context) error {
 	return nil
 }
 
+func (o *JobRunHistoricalDataAnalyzerOptions) getAlertData(ctx context.Context) []jobrunaggregatorapi.HistoricalData {
+	var allKnownAlerts []*jobrunaggregatorapi.KnownAlertRow
+	var newHistoricalData []jobrunaggregatorapi.HistoricalDataWithSetters
+
+	newHistoricalData, err := o.ciDataClient.ListAlertHistoricalData(ctx)
```

Comment:
This should now be  []*jobrunaggregatorapi.AlertHistoricalDataRow and you can use it in your innner loop below